src/main/io/vtx_smartaudio.c

```diff
@@ -0,0 +1,1271 @@
+    uint16_t orfreq;
+} smartAudioDevice_t;
+
+static smartAudioDevice_t saDevice = {
+    .version = 0,
+    .chan = -1,
+    .power = -1,
+    .mode = 0,
+    .freq = 0,
+    .orfreq = 0,
+};
+
+static smartAudioDevice_t saDevicePrev = {
+    .version = 0,
+};
+
+// XXX Possible compliance problem here. Need LOCK/UNLOCK menu?
+static uint8_t saLockMode = SA_MODE_SET_UNLOCK; // saCms variable?
+
+// XXX Should be configurable by user?
+static bool saDeferred = true; // saCms variable?
+
+// Receive frame reassembly buffer
+#define SA_MAX_RCVLEN 11
+static uint8_t sa_rbuf[SA_MAX_RCVLEN+4]; // XXX delete 4 byte guard
+
+//
+// CRC8 computations
+//
+
+#define POLYGEN 0xd5
+
+static uint8_t CRC8(const uint8_t *data, const int8_t len)
+{
+    uint8_t crc = 0; /* start with 0 so first byte can be 'xored' in */
+    uint8_t currByte;
+
+    for (int i = 0 ; i < len ; i++) {
+        currByte = data[i];
+
+        crc ^= currByte; /* XOR-in the next input byte */
+
+        for (int i = 0; i < 8; i++) {
+            if ((crc & 0x80) != 0) {
+                crc = (uint8_t)((crc << 1) ^ POLYGEN);
+            } else {
+                crc <<= 1;
+            }
+        }
+    }
+    return crc;
+}
+
+
+static void saPrintSettings(void)
+{
+#ifdef SMARTAUDIO_DPRINTF
+    dprintf(("Current status: version: %d\r\n", saDevice.version));
+    dprintf(("  mode(0x%x): fmode=%s", saDevice.mode,  (saDevice.mode & 1) ? "freq" : "chan"));
+    dprintf((" pit=%s ", (saDevice.mode & 2) ? "on " : "off"));
+    dprintf((" inb=%s", (saDevice.mode & 4) ? "on " : "off"));
+    dprintf((" outb=%s", (saDevice.mode & 8) ? "on " : "off"));
+    dprintf((" lock=%s", (saDevice.mode & 16) ? "unlocked" : "locked"));
+    dprintf((" deferred=%s\r\n", (saDevice.mode & 32) ? "on" : "off"));
+    dprintf(("  chan: %d ", saDevice.chan));
+    dprintf(("freq: %d ", saDevice.freq));
+    dprintf(("power: %d ", saDevice.power));
+    dprintf(("pitfreq: %d ", saDevice.orfreq));
+    dprintf(("\r\n"));
+#endif
+}
+
+static int saDacToPowerIndex(int dac)
+{
+    int idx;
+
+    for (idx = 0 ; idx < 4 ; idx++) {
+        if (saPowerTable[idx].valueV1 <= dac)
+            return(idx);
+    }
+    return(3);
+}
+
+//
+// Autobauding
+//
+
+#define SMARTBAUD_MIN 4800
+#define SMARTBAUD_MAX 4950
+uint16_t sa_smartbaud = SMARTBAUD_MIN;
+static int sa_adjdir = 1; // -1=going down, 1=going up
+static int sa_baudstep = 50;
+
+#define SMARTAUDIO_CMD_TIMEOUT    120
+
+static void saAutobaud(void)
+{
+    if (saStat.pktsent < 10)
+        // Not enough samples collected
+        return;
+
+#if 0
+    dprintf(("autobaud: %d rcvd %d/%d (%d)\r\n",
+        sa_smartbaud, saStat.pktrcvd, saStat.pktsent, ((saStat.pktrcvd * 100) / saStat.pktsent)));
+#endif
+
+    if (((saStat.pktrcvd * 100) / saStat.pktsent) >= 70) {
+        // This is okay
+        saStat.pktsent = 0; // Should be more moderate?
+        saStat.pktrcvd = 0;
+        return;
+    }
+
+    dprintf(("autobaud: adjusting\r\n"));
```

Comment:
You are right. I should disable the SMARTAUDIO_DPRINTF at the beginning of io/vtx_smartaudio.c for pushing onto the GitHub.